it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,306 @@
+package com.google.cloud.teleport.it.datastream;
+
+import com.google.cloud.datastream.v1.*;
+import com.google.protobuf.FieldMask;
+import org.apache.arrow.util.VisibleForTesting;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Default class for implementation of {@link DatastreamResourceManager} interface.
+ */
+public final class DefaultDatastreamResourceManager implements DatastreamResourceManager {
+    private static final String AVRO_FILE_FORMAT = "avro";
+    private static final String JSON_FILE_FORMAT = "json";
+    private static final String FIELD_STATE = "state";
+    private final DatastreamClient datastreamClient;
+    private final String location;
+    private final String projectId;
+    private final Map<String, ArrayList<String>> allowedTables;
+    private final String gcsBucketName;
+    private final String gcsRootPath;
+    private final String destinationOutputFormat = AVRO_FILE_FORMAT;
+    private final Source source;
+    private final Destination destination;
+    private final Set<String> createdStreamIds;
+    private final Set<String> createdConnectionProfileIds;
+
+    enum Source {
+        ORACLE,
+        MYSQL,
+        POSTGRESQL,
+    }
+
+    enum Destination {
+        GCS,
+        BQ,
+    }
+
+    private DefaultDatastreamResourceManager(Builder builder) {
+        this(
+                builder.datastreamClient,
+                builder.location,
+                builder.projectId,
+                builder.allowedTables,
+                builder.gcsBucketName,
+                builder.gcsRootPath,
+                builder.source,
+                builder.destination);
+    }
+
+    @VisibleForTesting
+    public DefaultDatastreamResourceManager(
+            DatastreamClient datastreamClient,
+            String location,
+            String projectId,
+            Map<String, ArrayList<String>> allowedTables,
+            String gcsBucketName,
+            String gcsRootPath,
+            Source source,
+            Destination destination) {
```

Comment:
I didn't go through every single one, but most of these variables are only used once, in which case they do not need to be stored as instance variables. `gcsBucketName` for example is only used when creating the destination connection profile, so it can be given as a parameter to the `createDestinationConnectionProfile()` method. `projectId`, however, is used in nearly every method, so it makes sense to be given to the constructor and stored as an instance variable.

Reply:
done. I assume the constants(e.g `AVRO_FILE_FORMAT`) should remain as it is despite being only used once